src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public <T extends Machine> List<Machine> getAll(Class<T> type) {
+        MachineProducer machineProducer = null;
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
```

Comment:
```suggestion
            return new BulldozerProducer().get();
```
the same for other ifs